src/main/java/vg/civcraft/mc/civduties/ModeManager.java

```diff
@@ -69,16 +69,17 @@ public boolean disableDutyMode(Player player, Tier tier) {
 			return false;
 		}
 		NBTCompound input = db.getPlayerData(player.getUniqueId()).getData();
-		// Inform the client the gamemode was changed to fix graphical issues on the
-		// client side
-		player.setGameMode(getGameModeByValue(input.getInteger("playerGameType")));
 		// Teleport the players using the bukkit api to avoid triggering nocheat
 		// movement detection
 		double [] location = input.getDoubleArray("Pos");
 		UUID worldUUID = UUID.fromString(input.getString("worldUUID"));
 		Bukkit.getScheduler().scheduleSyncDelayedTask(CivDuties.getInstance(), () -> {
 			player.teleport(new Location(Bukkit.getWorld(worldUUID), location [0], location[1], location[2]));
 		}, 3L);
+		// Inform the client the gamemode was changed to fix graphical issues on the
```

Comment:
Right above this the teleport is scheduled, meaning its gonna happen on the next tick. Moving the game mode change below it isn't gonna change anything, it's still gonna happen before the next tick. You need to move it into the scheduled task

Reply:
@Maxopoly Not sure what the methods below the gamemode change are doing - after adding the gamemode change into the scheduled task would I also need to schedule `cPlayer.getHandle().loadData(input.getRAW());` or `db.removePlayerData(player.getUniqueId());` ?

Reply:
Hmm, well actually, you might want to teleport twice in that case. It's a bit weird, because below we're doing something really hacky to restore the players player data and need to teleport one tick later to fix their location.